admin/class-my-yoast-proxy.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin
+ */
+
+/**
+ * @class WPSEO_MyYoast_Proxy Loads the MyYoast proxy.
+ *
+ * This class registers a proxy page on `admin.php`. Which is reached with the `page=PAGE_IDENTIFIER` parameter.
+ * It will read external files and serves them like they are located locally.
+ */
+class WPSEO_My_Yoast_Proxy implements WPSEO_WordPress_Integration {
+
+	/**
+	 * The page identifier used in WordPress to register the MyYoast proxy page.
+	 *
+	 * @var string
+	 */
+	const PAGE_IDENTIFIER = 'wpseo_myyoast_proxy';
+
+	/**
+	 * The cache control's max age. Used in the header of a successful proxy response.
+	 *
+	 * @var int
+	 */
+	const CACHE_CONTROL_MAX_AGE = 86400;
+
+	/**
+	 * Registers the hooks when the user is on the right page.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		if ( ! $this->is_proxy_page() ) {
+			return;
+		}
+
+		// Register the page for the proxy.
+		add_action( 'admin_menu', array( $this, 'add_proxy_page' ) );
+		add_action( 'admin_init', array( $this, 'handle_proxy_page' ) );
+	}
+
+	/**
+	 * Registers the proxy page. It does not actually add a link to the dashboard.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return void
+	 */
+	public function add_proxy_page() {
+		add_dashboard_page( '', '', 'read', self::PAGE_IDENTIFIER, '' );
+	}
+
+	/**
+	 * Renders the requested proxy page and exits to prevent the WordPress UI from loading.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return void
+	 */
+	public function handle_proxy_page() {
+		$this->render_proxy_page();
+		exit;
+	}
+
+	/**
+	 * Renders the requested proxy page.
+	 *
+	 * This is separated from the exits to be able to test it.
+	 *
+	 * @return void
+	 */
+	public function render_proxy_page() {
+		$proxy_options = $this->determine_proxy_options();
+		if ( $proxy_options === array() ) {
+			// Do not accept any other file than implemented.
+			$this->set_header( 'HTTP/1.0 501 Requested file not implemented' );
+			return;
+		}
+
+		if ( $this->load_url_directly() ) {
+			$this->set_header( 'Content-Type: ' . $proxy_options['content_type'] );
+			$this->set_header( 'Cache-Control: max-age=' . self::CACHE_CONTROL_MAX_AGE );
+
+			/*
+			 * If an error occurred, fallback to the next proxy method (`wp_remote_get`).
+			 * Otherwise, we are done here.
+			 */
+			if ( $this->load_url( $proxy_options['url'] ) ) {
```

Comment:
Instead of doing this, you might consider in throwing an exception in `load_url`. Then you can do this:

```php
try {
   $this->load_url( $proxy_options['url'];
}
catch( Exception $e ) {
   // Do some fallback magic - like rendering error page.
} 
```